app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt

```diff
@@ -78,18 +79,26 @@ class SelfUserProfileViewModel @Inject constructor(
 
     private fun observeEstablishedCall() {
         viewModelScope.launch {
-            observeEstablishedCalls().map { it.isNotEmpty() }.distinctUntilChanged().collect {
-                userProfileState = userProfileState.copy(isUserInCall = it)
-            }
+            observeEstablishedCalls()
+                .map { it.isNotEmpty() }
+                .distinctUntilChanged()
+                .flowOn(dispatchers.io())
+                .shareIn(this, SharingStarted.WhileSubscribed(1))
```

Comment:
do we need to convert the flow to hot SharedFlow since it will only bbe observed in this viewModel ?

Reply:
Actually nope, I changed a bit this code, to just switch the dispatcher for use case call.
I think we should later take the approach to switch in use cases, rather than here. We need to revisit that conversation again in the collective =)